storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogSegmentMetadata.java

```diff
@@ -100,7 +100,16 @@ public RemoteLogSegmentMetadata(RemoteLogSegmentId remoteLogSegmentId,
         this.remoteLogSegmentId = Objects.requireNonNull(remoteLogSegmentId, "remoteLogSegmentId can not be null");
         this.state = Objects.requireNonNull(state, "state can not be null");
 
+        if (startOffset < 0) {
+            throw new IllegalArgumentException(
+                String.format("Unexpected start offset = %d. StartOffset for a remote segment cannot negative", startOffset));
```

Comment:
Why `String.format` is used here and in the below exception cases?

Reply:
No specific reason. I think it's visually more clear to read the log string part together instead of adding a `+`. But I don't have strong opinion about it. I will change to append if that's what you prefer.

Reply:
this is fixed in latest commit